AlphaWallet/UI/TokenObject+UI.swift

```diff
@@ -54,63 +54,24 @@ class TokenImageFetcher {
     private static var subscribables: ThreadSafeDictionary<String, Subscribable<TokenImage>> = .init()
     private let queue: DispatchQueue = .global()
 
-    private func programmaticallyGenerateIcon(for contractAddress: AlphaWallet.Address, server: RPCServer, symbol: String) -> TokenImage {
-        let i = [TokenObject.numberOfCharactersOfSymbolToShowInIcon, symbol.count].min()!
+    private static func programmaticallyGenerateIcon(for contractAddress: AlphaWallet.Address, server: RPCServer, symbol: String) -> TokenImage? {
+        guard let i = [TokenObject.numberOfCharactersOfSymbolToShowInIcon, symbol.count].min() else { return nil }
         let symbol = symbol.substring(to: i)
         return (image: programmaticallyGeneratedIconImage(for: contractAddress, server: server), symbol: symbol, isFinal: false)
     }
 
     //Relies on built-in HTTP/HTTPS caching in iOS for the images
     func image(forToken tokenObject: TokenObject) -> Subscribable<TokenImage> {
-        let subscribable: Subscribable<TokenImage>
-        let key = "\(tokenObject.contractAddress.eip55String)-\(tokenObject.server.chainID)"
-        if let sub = Self.subscribables[key] {
-            subscribable = sub
-            if let value = sub.value, value.isFinal {
```

Comment:
Can you help to check? I think this is the version which should be copied into: `image(contractAddress:server:name:type:balance:)`. Because it contains the fix in #3112